src/ioctls/vm.rs

```diff
@@ -246,6 +255,28 @@ impl VmFd {
         Ok(new_vcpu(vcpu, kvm_run_ptr))
     }
 
+    /// Creates an emulated device in the kernel.
+    pub fn create_device(&self, device: &mut kvm_create_device) -> Result<DeviceFd> {
+        let ret = unsafe { ioctl_with_ref(self, KVM_CREATE_DEVICE(), device) };
+        if ret == 0 {
+            Ok((new_device(unsafe { File::from_raw_fd(device.fd as i32) })))
+        } else {
+            return Err(io::Error::last_os_error());
+        }
+    }
+
+    /// This queries the kernel for the preferred target CPU type.
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    pub fn get_preferred_target(&self, kvi: &mut kvm_vcpu_init) -> Result<()> {
```

Comment:
We should try to name methods as close as possible to the KVM ioctls that they provide wrappers for. Usually the names of all methods strip the `KVM_` prefix and use lower case. For this one it would be arm_preferred_target.